### DISCUSSION TITLE
Extending default theme

### DISCUSSION THREAD

**Discussion by stritti** [vuepress/core#393] [Q&A] [answered]
I tried to follow [cook book entry](https://vuepress.github.io/advanced/cookbook/extending-a-theme.html#extend-default-theme) to extend the default theme with footer, but I have nowhere found the footer after restart the dev-server.

I also tried to update `.vuepress/config.js` to point to theme `vuepress-theme-local` but then the build task crashes. 

I added the two files by copy & paste from cook book for initial start:

- `.vuepress/theme/index.js`
- `.vuepress/theme/layouts/Layout.vue`

**Comment by meteorlxy** [comment, marked as answer]
> I also tried to update .vuepress/config.js to point to theme vuepress-theme-local

You need to point to the entry file of the local theme

**Reply by stritti** [reply]
Thanks @meteorlxy now I had success. Maybe you could extend the cook book by additional part?

### Update `.vuepress/config.js`

- add import: `const { path } = require('@vuepress/utils')`
- add `theme` property to config

  ```javascript
  const { path } = require('@vuepress/utils')
  module.exports = {
     [...]
     theme: path.resolve(__dirname, './theme'),
     [...]
  }
   ```

**Reply by meteorlxy** [reply]
Well, as the cookbook is under the **Advanced Guide**, I think users should have read the basic Guide:

https://vuepress.github.io/guide/theme.html#local-theme

**Reply by meteorlxy** [reply]
But yes, cookbook should be a more complete guide. We should add that snippet, too

**Reply by stritti** [reply]
> Well, as the cookbook is under the **Advanced Guide**, I think users should have read the basic Guide:
> 
> 
> 
> https://vuepress.github.io/guide/theme.html#local-theme

I read this, but thought extending the theme works without.
